tasks/task_qc_utils.wdl

```diff
@@ -91,6 +91,110 @@ task fastqc_se {
     maxRetries:   3
   }
 }
+task fastq_scan {
+
+  input {
+    File        read1
+    File        read2
+    String      read1_name = basename(basename(basename(read1, ".gz"), ".fastq"), ".fq")
+    String      read2_name = basename(basename(basename(read2, ".gz"), ".fastq"), ".fq")
+  }
+
+  command <<<
+    # capture date and version
+    date | tee DATE
+    fastq-scan -v | tee VERSION
+
+    # set cat command based on compression
+    if [[ "~{read1}" == *".gz" ]] ; then 
+      cat_reads="zcat"
+    else
+      cat_reads="cat"
+    fi
+    
+    # capture forward read stats
+    eval "${cat_reads} ~{read1}" | fastq-scan | tee ~{read1_name}_fastq-scan.json >(jq .qc_stats.read_total > READ1_SEQS)
+    read1_seqs=$(cat READ1_SEQS)
+    eval "${cat_reads} ~{read2}" | fastq-scan | tee ~{read2_name}_fastq-scan.json >(jq .qc_stats.read_total > READ2_SEQS)
+    read2_seqs=$(cat READ2_SEQS) 
+      
+    # capture number of read pairs
+    if [ ${read1_seqs} == $read2_seqs]; then
+      read_pairs=${read1_seqs}
+    else
+      read_pairs="Uneven pairs: R1=${read1_seqs}, R2=${read2_seqs}"
+    fi
+
+    echo $read_pairs | tee READ_PAIRS
+  >>>
+
+  output {
+    File       read1_fastq_scan_report = "~{read1_name}_fastq-scan.json"
+    File       read2_fastq_scan_report = "~{read2_name}_fastq-scan.json"
+    Int        read1_seq = read_string("READ1_SEQS")
+    Int        read2_seq = read_string("READ2_SEQS")
+    String     read_pairs = read_string("READ_PAIRS")
+    String     version = read_string("VERSION")
+    String     pipeline_date = read_string("DATE")
+  }
+
+  runtime {
+    docker:       "quay.io/staphb/fastq-scan:0.4.3"
+    memory:       "2 GB"
+    cpu:          2
+    disks:        "local-disk 100 SSD"
+    preemptible:  0
+    maxRetries:   3
+  }
+}
+task fastq_scan_se {
+
+  input {
+    File        read1
+    String      read1_name = basename(basename(basename(read1, ".gz"), ".fastq"), ".fq")
+  }
+
+  command <<<
+    # capture date and version
+    date | tee DATE
+    fastq-scan -v | tee VERSION
```

Comment:
Do you want just the version number or `fastq-scan 0.4.3`?

If just version, something like `sed 's/fastq-scan //'` I think will do it

example in bactopia: https://github.com/bactopia/bactopia/blob/master/modules/local/bactopia/qc_reads/main.nf#L258

Reply:
I like having the entire `fastq-scan 0.4.3` to stay in-line with how other software versions are output
<img width="871" alt="Screen Shot 2022-01-28 at 3 31 44 PM" src="https://user-images.githubusercontent.com/20211421/151635725-be9118bf-34e4-4f8c-b4a5-15c0cd599fae.png">

Reply:
Whoa, wait. Looks like v0.4.1 is within the staphb/fastq-scan:0.4.3 docker image

Reply:
Ah, just looks like the version wasn't updated in [the 0.4.3 release](https://github.com/rpetit3/fastq-scan/blob/432f86d87ff9e44498734769fc133c8ab0cdd290/fastq-scan.cpp#L11)

Reply:
I see you're using 0.4.4 in biocontainer, but can't find the associated docker container image. If one is available on quay, I should be able to just switch the default to make sure these match

Reply:
Found the biocontainer image on quay and updated things on most recent commit

Reply:
Shoot. No jq in the biocontainer image. I'm going to make a new staphb docker-build with fastq-scan 0.4.4 and jq

Reply:
I added `jq` to the bioconda recipe. `quay.io/biocontainers/fastq-scan:0.4.4--h7d875b9_1` should do it for you if you want to use 0.4.4

Reply:
Thank you! Just put in the [PR for staphb/docker-build](https://github.com/StaPH-B/docker-builds/pull/286)s as well for @kapsakcj to review, but will switch to bioconda image for now